src/main/java/adris/altoclef/tasks/misc/RepairToolTask.java

```diff
@@ -0,0 +1,197 @@
+
+    private ItemTarget[] _toRepair;
+    private boolean _finished;
+    private final TimerGame _throwTimer = new TimerGame(0.5);
+
+    public RepairToolTask(ItemTarget... toRepair) {
+        _toRepair = toRepair;
+    }
+    public RepairToolTask() { //If this task is called without itemtarget, repair anything we can
+        this(
+            new ItemTarget(ItemHelper.NETHERITE_ARMORS),
+            new ItemTarget(ItemHelper.NETHERITE_TOOLS),
+            new ItemTarget(Items.ELYTRA),
+            new ItemTarget(ItemHelper.DIAMOND_ARMORS),
+            new ItemTarget(ItemHelper.DIAMOND_TOOLS),
+            new ItemTarget(ItemHelper.IRON_ARMORS),
+            new ItemTarget(ItemHelper.IRON_TOOLS),
+            new ItemTarget(ItemHelper.GOLDEN_ARMORS),
+            new ItemTarget(ItemHelper.GOLDEN_TOOLS),
+            new ItemTarget(ItemHelper.STONE_TOOLS),
+            new ItemTarget(ItemHelper.LEATHER_ARMORS),
+            new ItemTarget(ItemHelper.WOODEN_TOOLS),
+            new ItemTarget(Items.FISHING_ROD),
+            new ItemTarget(Items.FLINT_AND_STEEL),
+            new ItemTarget(Items.CARROT_ON_A_STICK),
+            new ItemTarget(Items.SHEARS),
+            new ItemTarget(Items.BOW),
+            new ItemTarget(Items.SHIELD),
+            new ItemTarget(Items.TRIDENT),
+            new ItemTarget(Items.CROSSBOW),
+            new ItemTarget(Items.WARPED_FUNGUS_ON_A_STICK),
+            new ItemTarget(Items.BOW)
+        );
+    }   
+    @Override
+    protected void onStart(AltoClef mod) {
+        _throwTimer.reset();
+        _toRepair = Arrays.stream(_toRepair).filter(target -> needRepair(mod, target)).toArray(ItemTarget[]::new); //Filter out items type we can't repair
+        Optional<ItemTarget> ItemTargetOPTRepair = Arrays.stream(_toRepair).findFirst(); //Get the first element in our list
+        if (!ItemTargetOPTRepair.isPresent()) {
+            Debug.logWarning("There is nothing to repair");
+        }
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        //We start this task by filtering out every item type that we can't repair :
+        //All items without mending or with no damage
+        _toRepair = Arrays.stream(_toRepair).filter(target -> needRepair(mod, target)).toArray(ItemTarget[]::new);
+        
+        //After that, we get the first item type to repair on the list
+        Optional<ItemTarget> ItemTargetOPTRepair = Arrays.stream(_toRepair).findFirst();
+        
+        if (ItemTargetOPTRepair.isPresent()) { //If the list is not empty
+            ItemTarget ItemTargetRepair = ItemTargetOPTRepair.get(); //We get the (real) first item on the list
+            
+            List<Slot> SlotRepair = mod.getItemStorage().getSlotsWithItemPlayerInventory(false, ItemTargetRepair.getMatches()); //And we get a list of every slot with that item
+            
+            Slot SlotRepairTarget = new PlayerSlot(-1); //Placeholder slot, will never get used if not replaced
```

Comment:
Why not use null as a placeholder instead? -1 can be mistaken for the cursor slot.